services/api/pepc_boost_test.go

```diff
@@ -206,6 +247,110 @@ func TestIsTxWEthDaiSwap(t *testing.T) {
 	}
 }
 
+func TestBaseTraceChecks(t *testing.T) {
+	_, _, backend := startTestBackend(t, common.EthNetworkGoerli)
+
+	cases := []struct {
+		description    string
+		callTrace      common.CallTrace
+		isTraceCorrect bool
+	}{
+		{
+			description: "Call to smart contract",
+			callTrace: common.CallTrace{
+				To: nil,
+			},
+			isTraceCorrect: false,
+		},
+		{
+			description: "Call type is STATICCALL",
+			callTrace: common.CallTrace{
+				Type: "STATICCALL",
+			},
+			isTraceCorrect: false,
+		},
+		{
+			description: "Call input is less then 4 bytes",
+			callTrace: common.CallTrace{
+				Input: []byte{0x01, 0x02, 0x03},
+			},
+			isTraceCorrect: false,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.description, func(t *testing.T) {
+			res, err := backend.relay.BaseTraceChecks(c.callTrace)
+			require.NoError(t, err)
+			require.Equal(t, c.isTraceCorrect, res)
+
+		})
+	}
+}
+
+// this is only for custom network
+func TestIsTraceEthUsdcSwap(t *testing.T) {
+	_, _, backend := startTestBackend(t, common.EthNetworkGoerli)
+
+	// Payload attributes
+	parentHash, feeRec, withdrawalsRoot, prevRandao, proposerPubkey, headSlot := GetTestPayloadAttributes(t)
+
+	prepareBackend(t, backend, headSlot, parentHash, feeRec, withdrawalsRoot, prevRandao, proposerPubkey, common.EthNetworkGoerli)
+
+	ethUsdcTraceContents := common.LoadFileContents(t, "../../testdata/traces/goerli/eth_usdc_trace.json")
+	ethUsdcTrace := new(common.CallTrace)
+	err := json.Unmarshal(ethUsdcTraceContents, ethUsdcTrace)
+	require.NoError(t, err)
+
+	pairToDifferentAddress := new(common.CallTrace)
+	err = json.Unmarshal(ethUsdcTraceContents, pairToDifferentAddress)
+	// some random address
+	pairToDifferentAddress.To = &uniswapV2Addr
+
+	ethUsdcTraceDifferentMethod := new(common.CallTrace)
+	err = json.Unmarshal(ethUsdcTraceContents, ethUsdcTraceDifferentMethod)
+	ethUsdcTraceDifferentMethod.Input = append([]byte("0x1234"), ethUsdcTrace.Input[4:]...)
+
+	cases := []struct {
+		description    string
+		callTrace      common.CallTrace
+		isTraceCorrect bool
+		requiredError  string
+	}{
+		{
+			description:    "valid trace",
+			callTrace:      *ethUsdcTrace,
+			isTraceCorrect: true,
+			requiredError:  "",
```

Comment:
seems like `requiredError` isnt used in any of the cases? worth removing if so

Reply:
true will remove!